schedule/schedule-appengine/src/main/client/jasify/admin/admin.routes.js

```diff
@@ -381,14 +381,7 @@
 
                         function allowed() {
                             if ($route.current.params.organizationId) {
-
-                                var dfd = $q.defer();
-
-                                Multipass.query($route.current.params.organizationId).then(function (result) {
-                                    dfd.resolve(angular.extend({items: []}, result));
-                                });
-
-                                return dfd.promise;
+                                return Multipass.query($route.current.params.organizationId);
                             } else {
                                 return {items: []};
```

Comment:
This is wrong as the API is now "broken".  When `$route.current.params.organizationId` is truthy we return a promise, and when it is falsy we return an object `{...}`.  This means that calling `allowed().then()` could raise a "no such function" when `...params.organizationID` is falsy.

Whenever you want to return a "then(able)"  promise just wrap it with a `$q.when()`.  From [the $q docs](https://code.angularjs.org/1.3.19/docs/api/ng/service/$q)

> Wraps an object that might be a value or a (3rd party) then-able promise into a $q promise. This is useful when you are dealing with an object that might or might not be a promise, or if the promise comes from a source that can't be trusted.


In this case

~~~javascript
if ($route.current.params.organizationId) {
    ...
} else {
    return $q.when( {item: []} );
}
~~~